### DISCUSSION TITLE
When attempting ssl with postgres

### DISCUSSION THREAD

**Discussion by bradyma1** [GoogleChrome/lighthouse-ci#510] [Q&A]
getting Unexpected token c in JSON at position 0 when starting docker instance - with

const fs = require('fs');

module.exports = {
  ci: {
    server: {
      storage: {
        storageMethod: 'sql',
        sqlDialect: 'postgres',
        sqlConnectionSsl: true,
        sqlConnectionUrl: process.env.DATABASE_URL,
        sqlDialectOptions: {
          ssl: {
            ca: fs.readFileSync('./certs/ca.crt', 'utf8'),
            key: fs.readFileSync('./certs/client.foo.key', 'utf8'),
            cert: fs.readFileSync('./certs/client.foo.crt', 'utf8'),
          },
        },
      },
    },
  },
};

Is this chrome?

**Comment by patrickhulce** [comment]
Thanks for filing @bradyma1! 

This isn't JSON. Are you putting this content into a `.json` file instead of a `.js` file?

**Comment by bradyma1** [comment]
yes.   I am guessing I need to use .js and .json

more detail...


const fs = require('fs');

module.exports = {
ci: {
server: {
storage: {
storageMethod: 'sql',
sqlDialect: 'postgres',
sqlConnectionSsl: true,
sqlConnectionUrl: postgres://postgres@1.2.3.4/lhci,
sqlDialectOptions: {
ssl: {
ca: fs.readFileSync('./certs/ca.crt', 'utf8'),
key: fs.readFileSync('./certs/key.key', 'utf8'),
cert: fs.readFileSync('./certs/client.crt', 'utf8'),
},
},
},
},
},
};

**Comment by bradyma1** [comment]
ok..got past that.....  starting with lhci@0.6.1 start: `lhci server -config=./lighthouse.js`

Is this still the format... and I am referencing lighthouse.js in the dockerfile..

Also is the run the same  now...or do I need more options?

**Comment by bradyma1** [comment]
It kind of like it ignoring my options...as it is not  .json.......

**Comment by bradyma1** [comment]
or do I need to convert these to .json....

**Comment by patrickhulce** [comment]
`lhci server -config=./lighthouse.js` will not use the config. `lhci server --config=./lighthouse.js` should use the config.

**Comment by bradyma1** [comment]
sorry typo.....hmmmm still coming back with Options...like it is  not seeing file..and the help show that .json is read in config.

**Comment by bradyma1** [comment]
actually I used ./lighthouserc.js......but I don't think that matters.

**Comment by patrickhulce** [comment]
Alright let's back up several steps then. Here's what an issue needs to have for us to be able to help :)

1. Clear reproduction steps.
2. Behavior you expected.
3. Behavior you observed.

Example:

**Steps**

```sh
$ docker run -it --entrypoint /bin/bash node:12-buster
$ npm install @lhci/server @lhci/cli
$ cat > config.json <<EOF
module.exports = {};
EOF
$ ./node_modules/.bin/lhci server --config=./config.json
```

**Expected**
Lighthouse uses my config file.

**Actual**
Lighthouse failed with `Unexpected token m in JSON at position 0`.

**Response to this Example Bug Report**
Change `config.json` to `config.js` now Lighthouse should fail with `Error: Cannot use sqlite without a database path` instead.

**Comment by bradyma1** [comment]
errno 1 Failed at the lhci@0.6.1 start script

unexpected token
code ELIFECYCLE
errno 1
Exit status 1
lhic@0.6.1 start: `lhci server --config=./lighthouserc.js

**Comment by bradyma1** [comment]
steps  

from your original example 
Dockerfile
From mode:12.10.0-alpine
VOLUME /out
EXPOSE 9001
WORKDIR /app

COPY package.json package-lock.json .npmrc lighthouserc.js /app/
COPY ca.crt client.crt key.key /certs/
RUN npm install

ADD static/fonts/roboto .node_modules/@lhci/server/dist
ADD static/material-icons ./node_modules/@lhci/server/dist

 all from your original example...

RUN sed
RUN sed
RUN sed

CMD [  "npm", ":start" ]

**Comment by bradyma1** [comment]
step 1 continued..also from original example with upgrade of lighthouseserver and lhci - 

package.json
"name': "lhci"
"scripts: {
     "start": lhci server --config=./lighthouserc.js"
},
"dependencies": {
    "@lhci/cli": "0.6.1",
    "@lhci/server": "0.6.1",
    "lighthouse": "^6.5.0",
    "pg": "^7.14.0",
    "pg-hstore": "^2.3.3"
}
}

**Comment by bradyma1** [comment]
Step 2 -
docker image build -t lhci .
Step 3
docker run -p 9001:9001 --mount='source=lhci-data,target=/data --detach lhci

errors listed above and docker instance does not come up.

Other things to note - also did docker volume create lhci-data

Excepted result, docker instance to come up... and ssl connect with postgres database

**Comment by bradyma1** [comment]
other info - 

 - added server certs to postgresql.conf server.crt, server.key
- turned on ssl=on - postgresql.conf
 - updated pg_hba.conf with hostssl all all 0.0.0.0/0 cert

**Comment by bradyma1** [comment]
another hint....even when running the command locally, I am geting the same error..almost like the --config does not recognize .js files

**Comment by patrickhulce** [comment]
I'm sorry @bradyma1 but I'm still not clear on what exact steps were taken, what very specific behavior you expect at a particular step, and what behavior happens at that particular step to believe there's a bug here. 

This information might be sufficient for other forums of helpers to look into but there's not enough here for us to act on with our limited time, so I've converted it to a discussion open to community instead.

**Comment by bradyma1** [comment]
ok......sorry if this is not clear......I am simply trying to convert a docker server instance formerly connecting to postgres non ssl to ssl...using your initial example and substituting the .json example with the new .js example provided above.  I cannot get the server to come up and connect to the postgres database in ssl mode with a client certificate or even attempt to connect to the database when using the .js file provided.

**Comment by bradyma1** [comment]
Also, whether attempting to connect to the postgres via the docker instance or via a simple lighthouserc.js.....I command is failing with

errno 1 Failed at the lhci@0.6.1 start script

unexpected token
code ELIFECYCLE
errno 1
Exit status 1
lhic@0.6.1 start: `lhci server --config=./lighthouserc.js
